test/testBreakCondition.cpp

```diff
@@ -345,6 +345,31 @@ TEST(PeriodicBox, low) {
 	EXPECT_DOUBLE_EQ(3, c.created.getPosition().z);
 }
 
+TEST(ReflectiveShell, inside) {
+	// Tests if the reflective boundaries place the particle back inside the shell
+	Vector3d center(0, 0, 0);
+	double radius = 100;
+	ReflectiveShell shell(center, radius);
+
+	Candidate c;
+	c.setCurrentStep(20);
+	c.previous.setPosition(Vector3d(80, 20, 30));
+	c.previous.setDirection(Vector3d(10, -1, -1));
+	c.current.setPosition(Vector3d(99.803, 18.0197, 28.0197
+));
+	c.current.setDirection(Vector3d(10, -1, -1));
+
+	shell.process(&c);
+
+	EXPECT_NEAR(89.9965, c.current.getPosition().x, 1e-4);
```

Comment:
Please leave a comment in the code on how you get to those expected values (or even calculate them with the trigonometric functions here).